### PR TITLE
HR-9 Adding Whole Openings Functionality

### PR DESCRIPTION
**Pull request by lufeva**
- View all Openings
- Add, Edit and Delete Opening

### CODE REVIEW

File: src/pages/Opening/Opening.jsx [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+        id
+      }
+      applications {
+        id
+      }
+    }
+  }
+`;
+
+const ADD_OPENING = gql`
```
Comment (yossely) [inline src/pages/Opening/Opening.jsx]: there is an `upsertOpening` mutation that can be used for adding and updating at the same time

File: src/pages/Opening/Opening.jsx [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+import React, { Component, Fragment } from 'react';
+import { Mutation, Query } from 'react-apollo';
+import { Button, DataTable, TableHeader, TableBody, TableRow, TableColumn, Snackbar, MenuButtonColumn, FontIcon } from 'react-md';
+import { withRouter } from 'react-router-dom';
+import OpeningDialog from './Dialog';
+import Loading from '../../components/Loading';
+import NumberFormat from 'react-number-format';
+import ConfirmDialog from '../../components/ConfirmDialog';
+
+class Opening extends Component {
```
Comment (yossely) [inline src/pages/Opening/Opening.jsx]: let's use `PropTypes` to specify the inputs for each component

File: src/pages/Opening/Opening.jsx [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+    this.setState({ showOpeningDialog: true });
+  };
+
+  hideDialog = () => {
+    this.setState({ showOpeningDialog: false });
+    this.opening = this.emptyOpening;
+  };
+
+  onClickEdit = openingToEdit => {
+    openingToEdit.steps = openingToEdit.steps.map(step => step.id);
```
Comment (yossely) [inline src/pages/Opening/Opening.jsx]: I don't know why, but when clicking _Edit_ to an opening the third time doesn't returns its steps nor the applications (the lists are empty) and if we click it one more time (4th) returns the error `TypeError: Cannot read property 'id' of undefined` in this line

File: src/pages/Opening/Opening.jsx (hunk 1 above)
Comment (lufeva) [reply src/pages/Opening/Opening.jsx]: Done

File: src/pages/Opening/Opening.jsx (hunk 2 above)
Comment (lufeva) [reply src/pages/Opening/Opening.jsx]: PropTypes Added

File: src/pages/Opening/Dialog.jsx [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+  }
+}
+
+const applicationAndStepsQuery = gql`
+  {
+    steps {
+      id
+      name
+    }
+    applications {
```
[the remark this answers is not in the document]
Comment (lufeva) [reply src/pages/Opening/Dialog.jsx]: Per chat, Applications are going to be Read-Only in the Openings view, so we avoid this kind of conflicts

[the remark this answers is not in the document]
Comment (lufeva) [reply src/pages/Opening/Dialog.jsx]: Done

File: src/pages/Opening/Opening.jsx (hunk 3 above)
Comment (lufeva) [reply src/pages/Opening/Opening.jsx]: Fixed. Thanks for testing

**Review by yossely — APPROVED** [review]

**Review by fvalencia — APPROVED** [review]

**Pull request merged**